device_doctor/lib/src/device.dart

```diff
@@ -4,22 +4,36 @@
 
 import 'dart:async';
 
+import 'android_device.dart';
 import 'health.dart';
 import 'ios_device.dart';
+import 'utils.dart';
 
 /// Discovers available devices and chooses one to work with.
 abstract class DeviceDiscovery {
   factory DeviceDiscovery(String deviceOs) {
     switch (deviceOs) {
       case 'ios':
+        for (String property in supportedIosProperties) {
```

Comment:
Where is supportedIosProperties coming from and why we need this logic?

Reply:
`supportedIosProperties` is defined in utils.dart, including `idevice_id`, idevicediagnostics`. These are the utilities we need for healthcheck and recovery.
From the bot_config when calling device_doctor, we need to inject these properties so that the device_doctor tool know where they are located.